### PR TITLE
Add better documentaion around variables and mixins

### PR DESCRIPTION
**Pull request by danhahn**
## PR Checklist

- Manually tested across supported browsers
  - [x] Chrome
  - [x] Firefox
  - [x] Safari
  - [x] IE11
- [x] Unit tests written (`common` at minimum)
- [x] PR has one of the `semver-` labels
- [ ] Two core team engineer approvals
- [x] One core team UX approval

### CODE REVIEW

File: src/styles/variables.less [hunk 1]
```diff
@@ -1,39 +1,62 @@
...
+// Standard inset shadow
+@shadow-inset: 0 10px 10px -10px @color-gray-30 inset;
+
+// Featerued color variables are intened to only be used for components that
+// have "states" to them like `info`, `waring`, `danger` etc.
+
+// `@featured-color-default-borderColor` should only be used for banner or
+// button or button like components.
+
+// Do not use `@featured-color-primary` insted use `@primary-color`
```
Comment (ogupte) [inline src/styles/variables.less]: s/insted/instead

File: src/styles/variables.less [hunk 2]
```diff
@@ -1,39 +1,62 @@
...
-@color-gray: #e3e3e3;
-@color-mediumGray: #c5c5c5;
-@color-darkGray: #333333;
+// Normailize transparent due to Safari bug
+@color-transparent: rgba(255, 255, 255, 0);
+
+// Standard inset shadow
+@shadow-inset: 0 10px 10px -10px @color-gray-30 inset;
+
+// Featerued color variables are intened to only be used for components that
```
Comment (ogupte) [inline src/styles/variables.less]: s/Featerued/Featured
s/intened/intended

File: src/styles/variables.less [hunk 3]
```diff
@@ -1,39 +1,62 @@
...
 @color-gray-10: fade(@color-black, 10%);
 @color-gray-25: fade(@color-black, 25%);
 @color-gray-30: fade(@color-black, 30%);
 
-// basic colors
-@color-lightGray: #f4f4f4;
-@color-gray: #e3e3e3;
-@color-mediumGray: #c5c5c5;
-@color-darkGray: #333333;
+// Normailize transparent due to Safari bug
```
Comment (ogupte) [inline src/styles/variables.less]: s/Normailize/Normalize

File: src/styles/variables.less [hunk 4]
```diff
@@ -1,39 +1,62 @@
...
+// Standard Background Color
+@color-backgroundColor: @color-lightGray;
+
+// General color
 @color-white: #fff;
 @color-black: #000;
 @color-primary: #2abbb0;
 
-@color-disabledText: tint(#333, 50%);
+// Genaral Grays
```
Comment (ogupte) [inline src/styles/variables.less]: s/Genaral/General

File: src/docs/containers/colors.jsx [hunk 5]
```diff
@@ -75,7 +107,11 @@ const colorList = [
 		]
 	},
 	{
-		category: 'featured colors',
+		category: 'Featured colors',
+		description: 'A featured color should only be used for a component that has ' +
+		'multible states like banners or buttons or button like components ' +
+		'(drop select).  Featured colors should not be consumed by most components ' +
+		'instead use the color variables defined above.',
```
Comment (ogupte) [inline src/docs/containers/colors.jsx]: should we use backticks or end slashes for multi lined strings?

File: src/styles/variables.less [hunk 6]
```diff
@@ -1,39 +1,62 @@
-// Base
-@borderRadius: 4px;
-
 // font properties
```
Comment (jondlm) [inline src/styles/variables.less]: I'd like to see some better grouping on this file. Something [ala bootstrap's approach](https://github.com/twbs/bootstrap/blob/d9a64a0281556240813de5bad7cd7ef2720b7a79/less/buttons.less#L1-L7)

**Pull request merged**